src/components/FetchError.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState } from "react";
+
+export const FetchError = ({ error }: { error: string}) => {
+  if (!error) {
+    return null
+  }
+
+  const isAuthError = error?.toLowerCase()?.includes('unauthorized');
+
+  return (
+    <div className="flex items-center justify-center p-12">
+      <p className="text-red-500">{error}</p>
+      {isAuthError && <AuthButton />}
+    </div>
+  )
+}
+
+const AuthButton = () => {
+  const [showModal, setShowModal] = useState(false);
+
+  return (
+    <>
+      <button
+        onClick={()=> setShowModal(true)}
+        className="hover:text-blue-800 text-gray-600 font-bold py-1 px-2 rounded mx-3"
+      >
+        Set auth token
+      </button>
+      {showModal && <AuthModal onClose={() => setShowModal(false)} /> }
+    </>
+  );
+}
+
+const AuthModal = ({ onClose }: { onClose: () => void }) => {
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    // @ts-expect-error - react doesn't know about form target keys
+    localStorage.setItem('debugger-token', e.target?.['auth-token'].value);
```

Comment:
!nit this can technically throw even if it's unlikely. i'll leave it to you on if you think it's worth handling.